modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java

```diff
@@ -102,64 +84,90 @@ public void testSendNotification() throws Exception {
+				user2.getEmailAddress()));
 
-		_addObjectEntry();
+		// Multiples emails for each main recipients, comma and space separator
 
-		notificationQueueEntries = ListUtil.sort(
-			notificationQueueEntryLocalService.getNotificationEntries(
-				NotificationConstants.TYPE_EMAIL,
-				NotificationQueueEntryConstants.STATUS_SENT),
-			Comparator.comparing(
-				notificationQueueEntry -> {
-					Map<String, Object> notificationRecipientSettingsMap =
-						NotificationRecipientSettingUtil.
-							getNotificationRecipientSettingsMap(
-								notificationQueueEntry);
+		_testSendNotification(
+			ListUtil.sort(
+				Arrays.asList(
+					user1.getEmailAddress(), user2.getEmailAddress())),
+			StringBundler.concat(
+				user1.getEmailAddress(), StringPool.COMMA_AND_SPACE,
+				user2.getEmailAddress()));
 
-					return String.valueOf(
-						notificationRecipientSettingsMap.get("to"));
-				}));
+		// Multiples emails for each main recipients, semicolon separator
 
-		Assert.assertEquals(
-			notificationQueueEntries.toString(), 2,
-			notificationQueueEntries.size());
+		_testSendNotification(
+			ListUtil.sort(
+				Arrays.asList(
+					user1.getEmailAddress(), user2.getEmailAddress())),
+			StringBundler.concat(
+				user1.getEmailAddress(), StringPool.SEMICOLON,
+				user2.getEmailAddress()));
 
-		List<String> expectedToEmailAddresses = ListUtil.sort(
-			Arrays.asList(user1.getEmailAddress(), user2.getEmailAddress()));
+		// Multiples emails for each main recipients, with terms and comma
+		// separator
 
-		_assertNotificationQueueEntry(
-			true, expectedToEmailAddresses.get(0),
-			notificationQueueEntries.get(0));
-		_assertNotificationQueueEntry(
-			true, expectedToEmailAddresses.get(1),
-			notificationQueueEntries.get(1));
+		_testSendNotification(
+			ListUtil.sort(
+				Arrays.asList(
+					user2.getEmailAddress(),
+					GetterUtil.getString(
+						childObjectEntryValues.get("emailTextObjectField")))),
+			"[%CURRENT_USER_EMAIL_ADDRESS%]," +
+				getTermName("emailTextObjectField"));
 
-		for (NotificationQueueEntry notificationQueueEntry :
-				notificationQueueEntries) {
+		// Multiples emails for each main recipients, with terms and comma and
+		// space separator
 
-			notificationQueueEntryLocalService.deleteNotificationQueueEntry(
-				notificationQueueEntry);
-		}
+		_testSendNotification(
+			ListUtil.sort(
+				Arrays.asList(
+					user2.getEmailAddress(),
+					GetterUtil.getString(
+						childObjectEntryValues.get("emailTextObjectField")))),
+			"[%CURRENT_USER_EMAIL_ADDRESS%], " +
+				getTermName("emailTextObjectField"));
+
+		// Multiples emails for each main recipients, with terms and semicolon
+		// separator
+
+		_testSendNotification(
+			ListUtil.sort(
+				Arrays.asList(
+					user2.getEmailAddress(),
+					GetterUtil.getString(
+						childObjectEntryValues.get("emailTextObjectField")))),
+			"[%CURRENT_USER_EMAIL_ADDRESS%];" +
+				getTermName("emailTextObjectField"));
```

Comment:
@pedro-oliveira446 maybe we can do only one test for checking separators, for example:

```java
StringBundler.concat(
				user1.getEmailAddress(), StringPool.COMMA,
				user2.getEmailAddress(), StringPool.COMMA_AND_SPACE,
				"[%CURRENT_USER_EMAIL_ADDRESS%];",
				user3.getEmailAddress())
```

Reply:
but I'm not sure, what do you think?

Reply:
@MarinhoFeliphe,
I think the way it is now, it's clearer what we're testing and it's the same way as requested in the bug. But, I'm not sure about that either.